src/VehicleSetup/FirmwareUpgradeController.cc

```diff
@@ -65,17 +80,25 @@ void FirmwareUpgradeController::startBoardSearch(void)
     _threadController->startFindBoardLoop();
 }
 
-void FirmwareUpgradeController::flash(FirmwareType_t firmwareType)
+void FirmwareUpgradeController::flash(AutoPilotStackType_t stackType,
+                                      FirmwareType_t firmwareType,
+                                      FirmwareVehicleType_t vehicleType)
 {
```

Comment:
But when you flash a flow (or Radio I think) you get here and stack is AutoPilotStackPX4, not PX4Flow which is what the data structures want. Hence the firmware lookup in the hash fails.